internal/e2e/artifact_test.go

```diff
@@ -19,3 +19,21 @@ func TestArtifactUploadDownload(t *testing.T) {
 		t.Errorf("Build state = %q, want %q", got, want)
 	}
 }
+
+// Test that an agent can upload and download artifact to/from a customer-managed S3 bucket
+func TestArtifactUploadDownload_CustomBucket(t *testing.T) {
+	ctx := t.Context()
+	tc := newTestCase(t, "artifact_custom_bucket.yaml")
```

Comment:
```suggestion
	tc := newTestCase(t, "artifact_custom_s3_bucket.yaml")
```

we are soon introducing GCP and Azure equivalent, might be worth to give a more specific name